components/org.wso2.carbon.identity.application.authenticator.adapter/src/test/java/org/wso2/carbon/identity/application/authenticator/adapter/util/EventContextBuilder.java

```diff
@@ -0,0 +1,141 @@
+ * under the License.
+ */
+
+package org.wso2.carbon.identity.application.authenticator.adapter.util;
+
+import org.wso2.carbon.identity.application.authentication.framework.context.AuthHistory;
+import org.wso2.carbon.identity.application.authentication.framework.context.AuthenticationContext;
+import org.wso2.carbon.identity.application.authentication.framework.model.AuthenticatedUser;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.Enumeration;
+import java.util.HashMap;
+import java.util.Map;
+
+import javax.servlet.http.HttpServletRequest;
+
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+public class EventContextBuilder {
+
+    Map<String, Object> eventContext = new HashMap<>();
+    public static final String SP_ID = "spId";
+    public static final String SP_NAME = "spName";
+    public static final String FLOW_ID = "flow-id";
+
+    /**
+     * Constructor for EventContextBuilder.
+     *
+     * @param authenticatedUser Authenticated user.
+     * @param headers           Headers.
+     * @param parameters        Parameters.
+     */
+    public EventContextBuilder(AuthenticatedUser authenticatedUser, String tenantDomain,
+                               Map<String, String> headers, Map<String, String> parameters,
+                               ArrayList<AuthHistory> authHistory) {
+
+        eventContext.put(AuthenticatorAdapterConstants.AUTH_REQUEST, buildAuthenticationRequest(headers, parameters));
+        eventContext.put(AuthenticatorAdapterConstants.AUTH_CONTEXT, buildAuthenticationContext(authenticatedUser,
+                tenantDomain, authHistory));
+    }
+
+    /**
+     * Get event context.
+     *
+     * @return Event context.
+     */
+    public Map<String, Object> getEventContext() {
+
+        return eventContext;
+    }
+
+    /**
+     * Set request.
+     *
+     * @param request Request.
+     */
+    public void setRequest(HttpServletRequest request) {
+
+        eventContext.put(AuthenticatorAdapterConstants.AUTH_REQUEST, request);
+    }
+
+    /**
+     * Set authentication context.
+     *
+     * @param context Authentication context.
+     */
+    public void setAuthenticationContext(AuthenticationContext context) {
+
+        eventContext.put(AuthenticatorAdapterConstants.AUTH_CONTEXT, context);
+    }
+
+    /**
+     * Set flow ID.
+     *
+     * @param flowId Flow ID.
+     */
+    public void setFlowId(String flowId) {
+
+        eventContext.put(AuthenticatorAdapterConstants.FLOW_ID, flowId);
+    }
+
+    public static ArrayList<AuthHistory> buildAuthHistory() {
+
+        ArrayList<AuthHistory> authHistory = new ArrayList<>();
+        AuthHistory authStep1 = new AuthHistory("authenticator-1", "LOCAL");
+        authHistory.add(authStep1);
+        AuthHistory authStep2 = new AuthHistory("authenticator-2", "GOOGLE");
+        authHistory.add(authStep2);
+        return authHistory;
+    }
```

Comment:
looks like hard coded

Reply:
This is class in tests package. Renamed the class.